docs/modules/ROOT/pages/how-to-guides/proc_hermetic-builds.adoc

```diff
@@ -36,11 +35,11 @@ spec:
 +
 [NOTE]
 ====
-* In a non-java application, that is a link:https://github.com/redhat-appstudio/build-definitions/blob/main/task/buildah/0.1/buildah.yaml[Buildah] task, adding the parameter mentioned above automatically isolates the build from the network, restricting it to only build from dependencies listed in your Git repository. If you need to pull in dependencies from a package manager for one of the xref:how-to-guides/proc_prefetching-dependencies-to-support-hermetic-build.adoc#supported-languages[supported languages], see xref:how-to-guides/proc_prefetching-dependencies-to-support-hermetic-build.adoc[Prefetching the package manager dependencies for the Hermetic build].
+* In a non-Java application that is a link:https://github.com/redhat-appstudio/build-definitions/blob/main/task/buildah/0.1/buildah.yaml[Buildah] task, adding the '**hermetic**' and '**true**' parameters automatically isolates the build from the network, which restricts it to building only from dependencies listed in your Git repository. If you need to pull in dependencies from a package manager for one of the xref:how-to-guides/proc_prefetching-dependencies-to-support-hermetic-build.adoc#supported-languages[supported languages], see xref:how-to-guides/proc_prefetching-dependencies-to-support-hermetic-build.adoc[Prefetching the package manager dependencies for the Hermetic build].
 
-* Do not add these parameters to the link:https://github.com/burrsutter/partner-catalog-stage/blob/e2ebb05ba8b4e842010710898d555ed3ba687329/.tekton/partner-catalog-stage-wgxd-pull-request.yaml#L87[`pipelineSpec.params`] section, as it should always display the default values for hermetic builds.
+* Do not add these parameters to the link:https://github.com/burrsutter/partner-catalog-stage/blob/e2ebb05ba8b4e842010710898d555ed3ba687329/.tekton/partner-catalog-stage-wgxd-pull-request.yaml#L87[`**pipelineSpec.params**`] section, as it should always display the default values for hermetic builds.
```

Comment:
Why not? What of a user wants hermetic builds by default for own pipeline.

Reply:
I know we've been tripped up in the past with the params across different sections. If they change it in `pipelinespec.params`, instead of `spec.params`, what does that change for their component? What about if they change it in both param sections?

Reply:
There are pipeline params and task params. Both are editable but the scope is different. That's up to user what to edit. I believe, we shouldn't deny editing of any params. Instead, we can warn, that there are task and pipeline params and user have to be careful to not to mismatch.

Reply:
In fact, we should point to Pipeline as Code / Tekton docs regarding pipeline definition.

Reply:
@chmeliik can you answer Mykola and Matt's questions?

Reply:
IIRC this section came out of a discussion in an earlier docs PR. Originally, the documentation suggested to enable hermetic builds by changing the _default_ value of the `hermetic` param. We thought that shouldn't be the suggested approach, the suggested approach should be to set the parameter value in the section intended for that purpose (i.e. `spec.params`).

Changing the default in `pipelineSpec.params` will also make the build hermetic, but will probably break assumptions that other systems are making. For example, Enterprise Contract will most likely think that the build wasn't hermetic, because the hermetic param wasn't set to `'true'`

Reply:
I wasn't aware of that EC bug. That's really confusing.

Reply:
@simonbaird Do you know if EC is sensitive to where the user sets hermetic to true in their pipeline definition? Based on https://github.com/enterprise-contract/ec-policies/blob/main/policy/release/hermetic_build_task.rego#L33 it looks to me like it checks the buildah task, and assuming it hasn't changed, the default pipeline config I have sets hermetic to false in `pipelineSpec.params`, does not include a hermetic value in `spec.params`, and the buildah task references `$(spec.hermetic)`. Guide for enabling it currently recommends adding hermetic: true to `spec.params`.

Reply:
@mattreid would you like me to wait for Simon's input or keep what Adam originally suggested?

Reply:
From https://redhat-internal.slack.com/archives/C031J4KBFME/p1705528600294819?thread_ts=1705525457.266349&cid=C031J4KBFME it doesn't sound like EC cares how the the param was set, as long as it was set in a way the task configuration is affected, the evidence they need will be available in the attestation for that taskrun.

Reply:
So IMO we don't need to explicitly discourage `pipelineSpec.params`, but we should still document `spec.params` as the way go